query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/LdapAttributeName.java

```diff
@@ -0,0 +1,54 @@
+package org.codice.ddf.admin.ldap.fields;
+
+import static org.codice.ddf.admin.ldap.commons.LdapMessages.invalidUserAttribute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codice.ddf.admin.api.report.ErrorMessage;
+import org.codice.ddf.admin.common.fields.base.scalar.StringField;
+
+public class LdapAttributeName extends StringField {
+
+    public static final String DEFAULT_FIELD_NAME = "attriName";
+
+    public static final String FIELD_TYPE_NAME = "LdapAttributeName";
+
+    public static final String DESCRIPTION = "The name of an attribute that holds information about an LDAP entry.";
+
+    public LdapAttributeName() {
+        super(DEFAULT_FIELD_NAME, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    public LdapAttributeName(String fieldName) {
+        super(fieldName, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    @Override
+    public List<ErrorMessage> validate() {
+        List<ErrorMessage> errors = super.validate();
+        if (!errors.isEmpty()) {
+            return errors;
+        }
+
+        errors.addAll(validate(this));
+        return errors;
+    }
+
+    // TODO: 7/7/17 - tbatie - This validate should be reformated once there is a generic way to create MapField objects that contain different value field.
+    public static List<ErrorMessage> validate (StringField attribute){
+        List<ErrorMessage> errors = new ArrayList<>();
+
+        if (attribute.getValue()!= null && !attribute.getValue().matches("^[xX][0-9a-fA-F]+[a-zA-Z]*$")
```

Comment:
We should compile these patterns once before we check for matches that way we can avoid compiling each time.